kolibri/plugins/document_epub_render/assets/src/views/index.vue

```diff
@@ -0,0 +1,233 @@
+        type: Object,
+      },
+    },
+    data: () => ({
+      isFullscreen: false,
+      progress: 0,
+      timeout: null,
+      totalPages: null,
+    }),
+    computed: {
+      fullscreenAllowed() {
+        return ScreenFull.enabled;
+      },
+      epubURL() {
+        return this.defaultFile.storage_url;
+      },
+      mimicFullscreen() {
+        return !this.fullscreenAllowed && this.isFullscreen;
+      },
+      targetTime() {
+        return this.totalPages * 30;
+      },
+      documentLoading() {
+        return this.progress !== 1;
+      },
+    },
+    watch: {
+      scrollPos: 'checkPages',
+      scale(newScale, oldScale) {
+        // Listen to changes in scale, as we have to rerender every visible page if it changes.
+        const noChange = newScale === oldScale;
+        const firstChange = oldScale === null;
+
+        if (!noChange && !firstChange && this.renderer) {
+          this.renderer.resize(this.defaultWidth * newScale, this.defaultHeight * newScale);
+        }
+      },
+    },
+    created() {
+      global.ePub = Epub;
+      this.book = new Epub(this.epubURL);
+      if (this.fullscreenAllowed) {
+        ScreenFull.onchange(() => {
+          this.isFullscreen = ScreenFull.isFullscreen;
+        });
+      }
+    },
+    mounted() {
+      this.renderer = this.book.renderTo(this.$refs.epubContainer, {
+        manager,
+        view: iFrameView,
+      });
+      this.renderer.display();
+    },
+    beforeDestroy() {
+      if (this.timeout) {
+        clearTimeout(this.timeout);
+      }
+      this.updateProgress();
+      this.$emit('stopTracking');
+    },
+    destroy() {
+      delete global.ePub;
+    },
+    methods: {
+      toggleFullscreen() {
+        if (this.fullscreenAllowed) {
+          ScreenFull.toggle(this.$refs.docViewer);
+        } else {
+          this.isFullscreen = !this.isFullscreen;
+        }
+      },
+      zoomIn: throttle(function() {
+        this.scale += scaleIncrement;
+      }, renderDebounceTime),
+      zoomOut: throttle(function() {
+        this.scale -= scaleIncrement;
+      }, renderDebounceTime),
+      nextPage() {
+        this.renderer.next();
+      },
+      prevPage() {
+        this.renderer.prev();
+      },
+      updateProgress() {
+        this.$emit('updateProgress', this.sessionTimeSpent / this.targetTime);
+      },
+    },
+    $trs: {
+      exitFullscreen: 'Exit fullscreen',
+      enterFullscreen: 'Enter fullscreen',
+    },
+    vuex: {
+      getters: {
+        sessionTimeSpent,
+      },
+    },
+  };
+
+</script>
+
+
+<style lang="stylus" scoped>
+
+  @require '~kolibri.styles.definitions'
+
+  $keen-button-height = 48px
+  $fullscreen-button-height = 36px
+  $page-padding = 5px
+
+  .doc-viewer
+    position: relative
+    width: 90%
+    margin-left: auto
+    margin-right: auto
+
+    &:fullscreen
+      width: 100%
+      height: 100%
+      min-height: inherit
+      max-height: inherit
+
+    &-mimic-fullscreen
```

Comment:
I know this was copied from another renderer, but I'd really like to ban this practice of concatenating together new class names from parent class names in stylus. It makes it much harder to figure out where class definitions are because you can't search for, e.g. `doc-viewer-mimic-fullscreen`